Provide go template to display in a concise way ClusterTestSuite (#22)
kyma-incubator/octopus: docs/kubectl-integration.md

# Integration with Kubectl

> Octopus provides integration with `kubectl` to simplify working with it's resources.

Reviewer: ```suggestion
Octopus provides integration with `kubectl` to simplify the work on its resources.
```

> ## Cluster Test Suite Template

Reviewer: ```suggestion
## ClusterTestSuite template
```

> Status of a `ClusterTestSuite` contains many information.

Reviewer: To get the full status of ClusterTestSuite, run: 
```
kubectl get cts -oyaml {suite mame}
```

However, the output of this status contains a lot of information, which can be overwhelming. To get more concise but still informative output, use the template stored in `kubectl/suite.template`. Run:

Output from following command:
```
kubectl get cts -oyaml {suite mame}
```
can be overwhelming. To get more concise but still informative output, you can use template stored in `kubectl/suite.template`.

> ```

Reviewer 2: IMO we need to provide the command for downloading this template. 
```
curl -LO https://raw.githubusercontent.com/kyma-incubator/octopus/master/kubectl/suite.template
```

 kubectl get cts {suite name} -ogo-template-file --template={path to template file}
```

> Output:

Reviewer: ```suggestion
The simplified output looks as follows:
```

```
Name:           {suite name}
Concurrency:    1.0
MaxRetries:     1.0
Count:          1.0
StartTime:      2019-04-06T12:56:11Z
CompletionTime: 2019-04-06T13:22:38Z
Condition:      Failed
Tests:

>     test-monitoring - + 

Reviewer 2: please describe what `- +` means

    test-events +
    test-ui ?
```

> Next to the test names, status of the test executions is displayed:

Reviewer: ```suggestion
Next to the test name, there is a symbol which specifies the status of a given test execution:
```

> # Integration with Kubectl

Reviewer: ```suggestion
# Integration with kubectl
```

> Octopus provides integration with `kubectl` to simplify the work on its resources.

Reviewer: ```suggestion
Octopus provides integration with kubectl to simplify work on its resources.
```

## ClusterTestSuite template

> To get the full status of ClusterTestSuite, run:

Reviewer: ```suggestion
You can get the full status of ClusterTestSuite by running:
```

```
kubectl get cts -oyaml {suite mame}
```
However, the output of this status contains a lot of information, which can be overwhelming. 
To get more concise but still informative output, use the template stored in `kubectl/suite.template`. 

> To download recent version of the template, execute the following command:

Reviewer: ```suggestion
To download the recent version of the template, run:
```

```
curl -LO https://raw.githubusercontent.com/kyma-incubator/octopus/master/kubectl/suite.template
```

> Then, set the `template` argument to point to the downloaded file:

Reviewer: ```suggestion
Then, set the `--template` argument to point to the downloaded file:
```

```
 kubectl get cts {suite name} -ogo-template-file --template={path to template file}
```

The simplified output looks as follows:
```
Name:           {suite name}
Concurrency:    1.0
MaxRetries:     1.0
StartTime:      2019-04-06T12:56:11Z
CompletionTime: 2019-04-06T13:22:38Z
Condition:      Succeeded
Tests:
    test-monitoring - + 
    test-events +
    test-ui ?
```

Next to the test name, there is a symbol which specifies the status of a given test execution:
- `+` - test execution passed
- `-` - test execution failed
- `?` - test execution is in progress or not yet scheduled

> In the above example, `test-monitoring` failed at the first time, then it was retried and finally it succeeded.  

Reviewer: ```suggestion
>**NOTE:** In the given example, `test-monitoring` failed at first, then it was retired, and finally it succeeded.  
```
